CodeTracker/CodeTrackerUnitTest/CodingTrackerInputTest.cs

```diff
@@ -0,0 +1,102 @@
+using CodeTracker;
+using Spectre.Console.Testing;
+namespace CodeTrackerUnitTest;
+
+[TestClass]
+public class CodingTrackerInputTest
+{
+    private readonly TestConsole _console;
+    private readonly Input _input;
+    public CodingTrackerInputTest()
+    {
+        _console = new TestConsole();
+        _input = new Input(_console);
+    }
+    [TestMethod]
+    [DataRow(" ")]
+    [DataRow("description")]
+    public void TestingDescriptionInput(string input)
+    {
+        // Arrange
+        if (!string.IsNullOrWhiteSpace(input))
+        {
+            _console.Input.PushTextWithEnter(input);  // Valid input case
+            var result = _input.GetDescription();
+            Assert.AreEqual(input, result);  // Verify the description matches the input
+        }
+        else
+        {
+            _console.Input.PushTextWithEnter(input);  // Invalid input case
+
+            // Act & Assert: Expect an exception due to invalid input
+            Assert.ThrowsException<InvalidOperationException>(() => _input.GetDescription());
+        }
+    }
+
+    [TestMethod]
+    [DataRow("test")]
+    [DataRow("")]
+    [DataRow("22-10-2023")]
+    public void TestDateInput(string input)
```

Comment:
🟠 This test mixes valid and invalid date formats, and it relies on `Console.SetIn`, making it inconsistent.

💡 Use a mock input source and focus only on testing the date validation logic.

🔧 Test separately for:
- Empty input (should return `DateTime.Today`).
- Invalid date format (should throw an exception).
- Valid date format (should return parsed date).